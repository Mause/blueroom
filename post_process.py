import json
from datetime import datetime

from icalendar import Calendar, Event

with open("out.json") as f:
    shows = json.load(f)

cal = Calendar()
for show in shows:
    for date in show["dates"]:
        event = Event()
        event.add("summary", show["title"])
        event.add("dtstart", datetime.fromisoformat(date[0]))
        event.add("dtend", datetime.fromisoformat(date[1]))
        #        event.add('location', show['url'])
        event.add("summary", show["url"])  # + '\n\n\n' + show['desc'])
        cal.add_component(event)


with open("output/dates.ical", "wb") as fh:
    fh.write(cal.to_ical())
