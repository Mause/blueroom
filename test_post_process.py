from datetime import datetime
from pathlib import Path

from syrupy.assertion import SnapshotAssertion

from post_process import process, tz

dt = datetime(2020, 1, 1, 12, 0, 0, tzinfo=tz)


def test_post_process(snapshot: SnapshotAssertion) -> None:
    output = process([], dt, output_filename=Path("output/dates.ics")).decode()

    assert output == snapshot


def test_post_process_event(snapshot: SnapshotAssertion) -> None:
    output = process(
        [
            {
                "item_hash": "123",
                "title": "Test Event",
                "url": "https://example.com",
                "desc": "This is a test event",
                "html_desc": "<p>This is a test event</p>",
                "dates": [
                    {
                        "start": "2020-01-01T14:00:00+08:00",
                        "end": "2020-01-01T16:00:00+08:00",
                        "venue": "Backrooms",
                    }
                ],
            }
        ],
        dt,
        output_filename=Path("output/dates.ics"),
    ).decode()

    assert output == snapshot
