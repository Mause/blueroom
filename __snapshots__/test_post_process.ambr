# serializer version: 1
# name: test_post_process
  '''
  BEGIN:VCALENDAR
  VERSION:2.0
  PRODID:-//Dates Calendar//mxm.dk//
  COLOR:yellow
  NAME:Dates Calendar
  REFRESH-INTERVAL;VALUE=DURATION:PT1H
  SOURCE:https://mause.me/blueroom/dates.ics
  X-PUBLISHED-TTL:PT1H
  X-WR-CALNAME:Dates Calendar
  X-WR-TIMEZONE:Australia/Perth
  BEGIN:VTIMEZONE
  TZID:Australia/Perth
  COMMENT:This timezone only works from 1970-01-01 to 2038-01-01.
  BEGIN:STANDARD
  DTSTART:19700101T000000
  TZNAME:AWST
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0800
  END:STANDARD
  BEGIN:DAYLIGHT
  DTSTART:19741027T030000
  RDATE:19831030T030000,19911117T030000,20061203T030000,20071028T030000,2008
   1026T030000
  TZNAME:AWDT
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0900
  END:DAYLIGHT
  BEGIN:STANDARD
  DTSTART:19750302T030000
  RDATE:19840304T030000,19920301T030000,20070325T030000,20080330T030000,2009
   0329T030000
  TZNAME:AWST
  TZOFFSETFROM:+0900
  TZOFFSETTO:+0800
  END:STANDARD
  END:VTIMEZONE
  END:VCALENDAR
  
  '''
# ---
# name: test_post_process_event
  '''
  BEGIN:VCALENDAR
  VERSION:2.0
  PRODID:-//Dates Calendar//mxm.dk//
  COLOR:yellow
  NAME:Dates Calendar
  REFRESH-INTERVAL;VALUE=DURATION:PT1H
  SOURCE:https://mause.me/blueroom/dates.ics
  X-PUBLISHED-TTL:PT1H
  X-WR-CALNAME:Dates Calendar
  X-WR-TIMEZONE:Australia/Perth
  BEGIN:VTIMEZONE
  TZID:Australia/Perth
  COMMENT:This timezone only works from 1970-01-01 to 2038-01-01.
  BEGIN:STANDARD
  DTSTART:19700101T000000
  TZNAME:AWST
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0800
  END:STANDARD
  BEGIN:DAYLIGHT
  DTSTART:19741027T030000
  RDATE:19831030T030000,19911117T030000,20061203T030000,20071028T030000,2008
   1026T030000
  TZNAME:AWDT
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0900
  END:DAYLIGHT
  BEGIN:STANDARD
  DTSTART:19750302T030000
  RDATE:19840304T030000,19920301T030000,20070325T030000,20080330T030000,2009
   0329T030000
  TZNAME:AWST
  TZOFFSETFROM:+0900
  TZOFFSETTO:+0800
  END:STANDARD
  END:VTIMEZONE
  BEGIN:VEVENT
  SUMMARY:Test Event (Tickets AVAILABLE)
  DTSTART;TZID=Australia/Perth:20200101T140000
  DTEND;TZID=Australia/Perth:20200101T160000
  DTSTAMP:20200101T040000Z
  UID:123 2020-01-01T14:00:00+08:00
  DESCRIPTION:https://example.com/\n\n<p>This is a test event</p>\n\nUpdated
   : 12:00PM\, January  1\, 2020
  LAST-MODIFIED:20200101T040000Z
  LOCATION:Backrooms
  END:VEVENT
  END:VCALENDAR
  
  '''
# ---
# name: test_unusual_dates
  '''
  BEGIN:VCALENDAR
  VERSION:2.0
  PRODID:-//Unusual_Dates Calendar//mxm.dk//
  COLOR:yellow
  NAME:Unusual_Dates Calendar
  REFRESH-INTERVAL;VALUE=DURATION:PT1H
  SOURCE:https://mause.me/blueroom/unusual_dates.ics
  X-PUBLISHED-TTL:PT1H
  X-WR-CALNAME:Unusual_Dates Calendar
  X-WR-TIMEZONE:Australia/Perth
  BEGIN:VTIMEZONE
  TZID:Australia/Perth
  COMMENT:This timezone only works from 1970-01-01 to 2038-01-01.
  BEGIN:STANDARD
  DTSTART:19700101T000000
  TZNAME:AWST
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0800
  END:STANDARD
  BEGIN:DAYLIGHT
  DTSTART:19741027T030000
  RDATE:19831030T030000,19911117T030000,20061203T030000,20071028T030000,2008
   1026T030000
  TZNAME:AWDT
  TZOFFSETFROM:+0800
  TZOFFSETTO:+0900
  END:DAYLIGHT
  BEGIN:STANDARD
  DTSTART:19750302T030000
  RDATE:19840304T030000,19920301T030000,20070325T030000,20080330T030000,2009
   0329T030000
  TZNAME:AWST
  TZOFFSETFROM:+0900
  TZOFFSETTO:+0800
  END:STANDARD
  END:VTIMEZONE
  BEGIN:VEVENT
  SUMMARY:Love Stories (Tickets SOLD_OUT)
  DTSTART;TZID=Australia/Perth:20250704T183000
  DTEND;TZID=Australia/Perth:20250704T192000
  DTSTAMP:20200101T040000Z
  UID:ab14eaecc5144650a99de68d3f64bbca 2025-07-04T18:30:00+08:00
  DESCRIPTION:https://blueroom.org.au/events/love-stories/\n\nUpdated: 12:00
   PM\, January  1\, 2020
  LAST-MODIFIED:20200101T040000Z
  LOCATION:Kaos Room
  END:VEVENT
  BEGIN:VEVENT
  SUMMARY:Love Stories (Tickets SELLING_FAST)
  DTSTART;TZID=Australia/Perth:20250705T140000
  DTEND;TZID=Australia/Perth:20250705T145000
  DTSTAMP:20200101T040000Z
  UID:ab14eaecc5144650a99de68d3f64bbca 2025-07-05T14:00:00+08:00
  DESCRIPTION:https://blueroom.org.au/events/love-stories/\n\nUpdated: 12:00
   PM\, January  1\, 2020
  LAST-MODIFIED:20200101T040000Z
  LOCATION:Kaos Room
  END:VEVENT
  END:VCALENDAR
  
  '''
# ---
